Example/IntegrationTests/Push/PushTests.swift

```diff
@@ -0,0 +1,157 @@
+        dappPushClient = DappPushClientFactory.create(metadata: AppMetadata(name: name, description: "", url: "", icons: [""]),
+                                                      logger: pushLogger,
+                                                      keyValueStorage: keyValueStorage,
+                                                      keychainStorage: keychain,
+                                                      networkInteractor: networkingInteractor)
+    }
+
+    func makeWalletClients()  {
+        let prefix = "🦋 Wallet: "
+        let (pairingClient, networkingInteractor, keychain, keyValueStorage) = makeClientDependencies(prefix: prefix)
+        let pushLogger = ConsoleLogger(suffix: prefix + " [Push]", loggingLevel: .debug)
+        walletPairingClient = pairingClient
+        let echoClient = EchoClientFactory.create(tenantId: "", clientId: "")
+        walletPushClient = WalletPushClientFactory.create(logger: pushLogger,
+                                                          keyValueStorage: keyValueStorage,
+                                                          keychainStorage: keychain,
+                                                          networkInteractor: networkingInteractor,
+                                                          pairingRegisterer: pairingClient,
+                                                          echoClient: echoClient)
+    }
+
+    override func setUp() {
+        makeDappClients()
+        makeWalletClients()
+    }
+
+    func testRequestPush() async {
+        let expectation = expectation(description: "expects to receive push request")
+
+        let uri = try! await dappPairingClient.create()
+        try! await walletPairingClient.pair(uri: uri)
+        try! await dappPushClient.request(account: Account.stub(), topic: uri.topic)
+
+        walletPushClient.requestPublisher.sink { (topic, request) in
+            expectation.fulfill()
+        }
+        .store(in: &publishers)
+        wait(for: [expectation], timeout: InputConfig.defaultTimeout)
+    }
+
+    func testWalletApprovesPushRequest() async {
+        let expectation = expectation(description: "expects dapp to receive successful response")
+
+        let uri = try! await dappPairingClient.create()
+        try! await walletPairingClient.pair(uri: uri)
+        try! await dappPushClient.request(account: Account.stub(), topic: uri.topic)
+
+        walletPushClient.requestPublisher.sink { [unowned self] (id, _) in
+
+            Task(priority: .high) { try! await walletPushClient.approve(id: id) }
+        }.store(in: &publishers)
+
+        dappPushClient.responsePublisher.sink { (id, result) in
+            guard case .success = result else {
+                XCTFail()
+                return
+            }
+            expectation.fulfill()
+        }.store(in: &publishers)
+
+        wait(for: [expectation], timeout: InputConfig.defaultTimeout)
+    }
+
+    func testWalletRejectsPushRequest() {
```

Comment:
Planning to implement in this PR?

Reply:
but it is not implemented in the engine yet